src/string.cr

```diff
@@ -4244,8 +4244,11 @@ class String
   # Raises an `ArgumentError` if `self` has null bytes. Returns `self` otherwise.
   #
   # This method should sometimes be called before passing a `String` to a C function.
-  def check_no_null_byte
-    raise ArgumentError.new("String contains null byte") if byte_index(0)
+  def check_no_null_byte(name = nil)
+    if byte_index(0)
+      name = "`#{name}` " if name
```

Comment:
To show that this refers to a name. This formatting is used in many other compiler and stdlib error messages.

Reply:
Weird, I never noticed that.

Reply:
I think it's not that common

Reply:
IMHO `Key contains null byte` reads way better than `String 'key' contains null byte`.

Reply:
Why not just `"#{name || "String"} contains null byte"`?

Reply:
I propose "Argument '#{name}' contains null byte".

Reply:
@asterite Do you mean `name` should be required?

Reply:
Perhaps `check_no_null_byte(what = "String")` then just `#{what} contains null byte`?

Reply:
I'm looking at the code, it always seems to be a check against an argument. To avoid repeating the argument name, maybe we can have:

```crystal
class String
  macro check_no_null_byte(name)
    raise ArgumentError.new("Argument '{{name}}' contains null byte") if {{name}}.byte_index(0)
    {{name}}
  end
end
```

Then it can be used like this:

```crystal
def some_method(arg)
  String.check_no_null_byte(arg)
  # do something
end
```

That way there's no string interpolation, and there's no need to repeat the argument name.

Reply:
This is bikeshedding and overengineering. The diff is fine as-is.

Reply:
@RX14 I was writing the same thing. All are valid options and it doesn't make a difference regarding ENV.
Anyone can send a PR later to refactor `check_no_null_byte` if you care enough.